### ISSUE TITLE
Does this project work on Mac M1 CPU?

### ISSUE THREAD

**Issue by IPL** [bes-dev/stable_diffusion.openvino#15]
I tried but always failed in below line of code:
  self.unet = self.core.compile_model(self._unet, device)
There is error when compiling unet, with error "node input index is out of range".
Does the error due to unsupported M1 CPU?
Thanks.

**Comment by bes-dev** [comment]
@IPL no, this project not works on Mac M1 CPU, only x86_64 compatible CPU

**Comment by JanzenMark** [comment]
@IPL By first installing the rust compiler with "brew install rust", then editing requirements.txt to delete the version number from numpy and switching openvino to openvino-arm I was able to get it to fully install on my M1. However, the diffusion is failing dnow due to lack of weights being installed. I haven't had time to debug that but hopefully this information helps you make progress on your system.

**Comment by JustinGuese** [comment]
even if it would work the speed would be horrible, as the m1 would emulate the x86_64 stuff

**Comment by tfmoraes** [comment]
@IPL and @JanzenMark see this https://replicate.com/blog/run-stable-diffusion-on-m1-mac

**Comment by JanzenMark** [comment]
Thanks @tfmoraes . Worked great! And fast too. Getting 1.5s / iteration.

**Comment by ilya-lavrenov** [comment]
@tfmoraes @IPL @JustinGuese you can try to use non-official OpenVINO ARM package https://pypi.org/project/openvino-arm with https://github.com/openvinotoolkit/openvino_contrib/tree/master/modules/arm_plugin underneath

**Comment by zackees** [comment]
If you are willing to run the docker version then try forcing docker to run in x86-64 emulation mode on the Mac M1:

[Dockerfile](https://github.com/bes-dev/stable_diffusion.openvino/blob/master/Dockerfile):

Change
```
FROM python:3.9.9-bullseye
```

To
```
FROM --platform=linux/amd64 python:3.9.9-bullseye
```

**Comment by ilya-lavrenov** [comment]
Since OpenVINO 2022.3, pip openvino package is available natively for Apple M1 / M2
https://pypi.org/project/openvino/2022.3.0/#files

**Comment by blinkguy182** [comment]
> even if it would work the speed would be horrible, as the m1 would emulate the x86_64 stuff

M1 can emulate any x86 code as if it's nothing, I don't think that is an issue at all.
